packages/database/src/modelClasses/ExternalDatabaseConnection.js

```diff
@@ -5,13 +5,17 @@
 
 import { getEnvVarOrDefault, requireEnv } from '@tupaia/utils';
 import knex from 'knex';
+import { types as pgTypes } from 'pg';
 
 import { DatabaseModel } from '../DatabaseModel';
 import { DatabaseType } from '../DatabaseType';
 import { TYPES } from '../types';
 
 const EXT_DB_CONNECTION_ENV_VAR_PREFIX = 'EXT_DB';
 
+pgTypes.setTypeParser(pgTypes.builtins.NUMERIC, parseFloat);
+pgTypes.setTypeParser(20, parseInt); // bigInt type to Integer
```

Comment:
Nice one @biaoli0! Any idea why this issue is occurring here, but not for `TupaiaDatabase`? Pretty sure we pass numeric data fine there...

Also, I saw this code in `TupaiaDatabase` that seems to be used to get around timezone issues. Might be of interest:
```
// turn off parsing of timestamp (not timestamptz), so that it stays as a sort of "universal time"
// string, independent of timezones, rather than being converted to local time
pgTypes.setTypeParser(pgTypes.builtins.TIMESTAMP, val => val);
```

Reply:
Yeah I can't tell but it is weird. Can we take it as a tech debt? I will create a ticket and can take a deep look during the upcoming sprint.

Reply:
Agreed, can u add a ticket into data-tables phase 3 about it 👍

Reply:
As discussed with Edwin, there is no type issue before because Tupaia considers all these types as a string. Since `pgTypes.setTypeParser` will setup globally, it currently introduces a bug for many places. I will work on a hotfix about it.